app/modules/encoder.c

```diff
@@ -0,0 +1,154 @@
+// Module encoders
+
+#include "module.h"
+#include "lauxlib.h"
+#include "lmem.h"
+#include "c_string.h"
+#define BASE64_INVALID '\xff'
+#define BASE64_PADDING '='
+#define ISBASE64(c) (unbytes64[c] != BASE64_INVALID)
+
+static const char b64[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+static char *toBase64 ( lua_State* L, const char *msg, size_t *len){
+  size_t i, n = *len;
+
+  if (!n)  // handle empty string case 
+    return NULL;
+
+  char* q, *out = (char*)luaM_malloc(L, (n + 2) / 3 * 4);
+  char bytes64[sizeof(b64)];
+  c_memcpy(bytes64, b64, sizeof(b64));   //Avoid lots of flash unaligned fetches
+  
+  for (i = 0, q = out; i < n; i += 3) {
+    int a = msg[i];
```

Comment:
I suspect that since msg is char *, this will do sign expansion and then bad things will happen. Probably want either to make msg an unsigned char * or add & 0xff